curriculum/challenges/english/25-front-end-development/workshop-superhero-application-form/680900675ae3d54ee19590d3.md

```diff
@@ -0,0 +1,424 @@
+---
+id: 680900675ae3d54ee19590d3
+title: Step 19
+challengeType: 0
+dashedName: step-19
+---
+
+# --description--
+
+Noticed you can submit the form without filling some of the inputs or even all of them? You should add a validation then.
+
+To do that, use the `disabled` attribute to disable the button if `heroName`, `realName`, or `powerSource` are falsy, or if the length of `powers` is `0`.
+
+Now try to submit the form without filling in all the inputs or leave out any of them.
+
+With that, you superhero application form is complete.
+
+# --hints--
+
+You should not modify the existing content of the button.
+
+```js
+assert.match(code, /<button\s+className\s*=\s*('|")submit-btn('|")\s*type\s*=\s*('|")submit('|")\s*/)
```

Comment:
```suggestion
assert.match(code, /<\s*button\s+className\s*=\s*('|")\s*submit-btn\s*\1\s*type\s*=\s*('|")\s*submit\s*\2/)
```

`\1` to match the first single/double quote. `\2` to match the next capture group.